pna_control/pna_control.py

```diff
@@ -114,7 +136,40 @@ def powersweep(startpower: float, endpower: float, numsweeps: int, centerf: floa
     file.write('POINTS: '+str(points)+'\n')
     file.close()
 
-    #run each sweep while increasing averages for each power
+    #run each sweep
     for i in sweeps:
-        getdata(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        get_data(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        # Iterate number of averages as power decreases
         averages = averages * ((10**(stepsize/10))**0.5)
+
+def name_datafile(outputfile,power,temp):
+    filename = outputfile+'_'+str(power)+'dB'+'_'+str(temp)+'mK'
+    filename = filename.replace('.','p')
+
+    return filename
+    
+def timestamp_folder(dir,meastype):
+    result = time.localtime(time.time())
+    output = str(result.tm_year)
+    if len(str(result.tm_mon)) < 2:
+        output = output + '0' + str(result.tm_mon)
+    else:
+        output = output + str(result.tm_mon)
+    if len(str(result.tm_mday)):
+        output = output + '0' + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
+    else:
+        output = output + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
+    
+    output = meastype+ '_' + output
+    output = output.replace('.','p')
```

Comment:
There are some cool built-in functions to handle these use cases. For example some of this can be handled by:
`now = time.strftime("%y%m%d_%H_%M_%S", time.localtime())`

Reply:
Awesome! I replaced the old timestamping with this. Thanks!